src/poprox_recommender/diversifiers/topic_calibration.py

```diff
@@ -1,38 +1,40 @@
+import logging
 from collections import defaultdict
 
 import numpy as np
 import torch as th
-
 from poprox_concepts import Article, ArticleSet, InterestProfile
-from poprox_recommender.topics import extract_general_topics, normalized_topic_count
+
+from poprox_recommender.rankers import Ranker
+from poprox_recommender.topics import (extract_general_topics,
+                                       normalized_topic_count)
 
 
 # Topic Calibration uses MMR
 # to rerank recommendations according to
 # topic calibration
-class TopicCalibrator:
-    def __init__(self, algo_params, num_slots):
+class TopicCalibrator(Ranker):
+    def __init__(self, algo_params, num_slots=10):
+        self.validate_algo_params(algo_params, ["theta"])
+
         # Theta term controls the score and calibration tradeoff, the higher
         # the theta the higher the resulting recommendation will be calibrated.
         # We choose the binary search way to start in the middle
-        self.theta = float(algo_params.get("theta", 0.5))
+        self.theta = float(algo_params.get("theta", 0.1))
         self.num_slots = num_slots
 
     def __call__(self, candidate_articles: ArticleSet, interest_profile: InterestProfile) -> ArticleSet:
         article_scores = th.sigmoid(th.tensor(candidate_articles.scores)).cpu().detach().numpy()
 
         topic_preferences: dict[str, int] = {}
 
+        for interest in interest_profile.onboarding_topics:
+            topic_preferences[interest.entity_name] = max(interest.preference - 1, 0)
+
         if interest_profile.click_topic_counts:
             for topic, click_count in interest_profile.click_topic_counts.items():
                 topic_preferences[topic] = click_count
```

Comment:
Since you have the onboarding selections to initialize the topic preferences, the click counts should probably be added instead of overriding the value from above: 

```suggestion
                topic_preferences[topic] += click_count
```